CMakeLists.txt

```diff
@@ -182,16 +184,17 @@ endforeach()
 file(GLOB TEMP "${CMAKE_SOURCE_DIR}/*.h")
 list(APPEND HEADERS ${TEMP})
 
-add_library(flint ${SOURCES})
+add_library(flint SHARED ${SOURCES})
```

Comment:
This makes it impossible to build static libraries if needed. There's an option `cmake -DBUILD_SHARED_LIBS=on` which would build a shared library. (You could change the default value of off in BUILD_SHARED_LIBS to on if you like)